scipy/stats/stats.py

```diff
@@ -8121,6 +8121,281 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
+
+    Without further assumption on the samples (eg, the nature of the underlying
+    distribution), the one-sided intervals are optimally tight.
+
+    References
+    ----------
+    .. [1] W. R. Thompson, "On Confidence Ranges for the Median and Other
+       Expectation Distributions for Populations of Unknown Distribution Form,"
+       The Annals of Mathematical Statistics, vol. 7, no. 3, pp. 122-128, 1936,
+       Accessed: Sep. 18, 2019. [Online]. Available:
+       https://www.jstor.org/stable/2957563.
+    .. [2] H. A. David and H. N. Nagaraja, "Order Statistics in Nonparametric
+       Inference" in Order Statistics, John Wiley & Sons, Ltd, 2005, pp.
+       159-170. Available:
+       https://onlinelibrary.wiley.com/doi/10.1002/0471722162.ch7.
+    .. [3] N. Hutson, A. Hutson, L. Yan, "QuantileNPCI: Nonparametric
+       Confidence Intervals for Quantiles," R package,
+       https://cran.r-project.org/package=QuantileNPCI
+
+
+    Examples
+    --------
+    >>> from scipy.stats import confint_quantile
+    >>> x = [2, 8, 3, 6, 4, 1, 5, 9, 7]
+    >>> confint_quantile(x, 0.5, 0.95)
+    (2, 8)
+
+    To compute a two-sided interval instead, use the `type` parameter.
+
+    >>> confint_quantile(x, 0.5, 0.99, type='two-sided')
+    (1, 9)
+
+    You can also pass the number of samples as argument (instead of the samples)
+    themselves. The returned values are then the indexes of the upper and lower
+    bounds for the confidence intervals. We obtain the same result as in the
+    first example.
+
+    >>> N = len(x)
+    >>> lb, ub = confint_quantile(N, 0.5, 0.95)
+    >>> # Since we pass an interger to the function, we get the CI indexes
+    >>> print('Indexes:  ',lb, ub)
+    Indexes:   1 7
+    
+    >>> # Don't forgot to sort your data if working with indexes!
+    >>> print('CI bounds:',sorted(x)[lb], sorted(x)[ub])
+    CI bounds: 2 8
+
+    Generally, the more samples, the narrower the confidence intervals.
+
+    >>> for N in [5,10,100,1000]:
+    ...     x = np.random.rand(N)
+    ...     lb, ub = confint_quantile(x, 0.5, 0.95)
+    ...     med = np.median(sorted(x))
+    ...     print('N=%i'%N)
+    ...     print('lower-bound  %0.2f' % lb)
+    ...     print('emp. median  %0.2f' % med)
+    ...     print('upper-bound  %0.2f' % ub)
+    N=5
+    lower-bound  0.39  #random
+    emp. median  0.48  #random
+    upper-bound  0.96  #random
+    N=10
+    lower-bound  0.11  #random
+    emp. median  0.35  #random
+    upper-bound  0.71  #random
+    N=100
+    lower-bound  0.46  #random
+    emp. median  0.54  #random
+    upper-bound  0.63  #random
+    N=1000
+    lower-bound  0.47  #random
+    emp. median  0.52  #random
+    upper-bound  0.53  #random
+
+    .. versionadded:: 1.6.4
+    """
+
+    ##
+    # Checking the inputs
+    #
+    # x can be either an integer or a one-dimensional array-like
+    if isinstance(x, int):
+        if x < 1:
+            raise ValueError(
```

Comment:
Please consider converting these to [F-strings](https://realpython.com/python-f-strings/). 
Triple-quotes for line continuation is not needed within parentheses. You can end the quote on one line, then begin and end on the next, and they will automatically be considered as part of the same string.